plutus-core/untyped-plutus-core/test/Generators.hs

```diff
@@ -0,0 +1,111 @@
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeOperators    #-}
+
+module Generators where
+
+import PlutusPrelude
+
+import PlutusCore (Name, nameString)
+import PlutusCore.Compiler.Erase
+import PlutusCore.Default
+import PlutusCore.Error (ParserErrorBundle)
+import PlutusCore.Generators
+import PlutusCore.Generators.AST as AST
+import PlutusCore.Parser (defaultUni, parseGen)
+import PlutusCore.Pretty
+import PlutusCore.Quote
+import UntypedPlutusCore.Core.Type
+import UntypedPlutusCore.Parser
+
+import Data.Text qualified as T
+
+import Hedgehog (property, tripping)
+import Test.Tasty
+import Test.Tasty.HUnit
+import Test.Tasty.Hedgehog
+
+import Flat qualified
+
+-- | A 'Program' which we compare using textual equality of names rather than alpha-equivalence.
+newtype TextualProgram a = TextualProgram { unTextualProgram :: Program Name DefaultUni DefaultFun a } deriving stock Show
+
+instance Eq a => Eq (TextualProgram a) where
+    (TextualProgram p1) == (TextualProgram p2) = compareProgram p1 p2
```

Comment:
Do you think we can replace all these with a derived read and eq instance?

Reply:
Yes, it's what I hinted at in [this](https://github.com/input-output-hk/plutus/pull/4738#discussion_r908718206) comment.